src/main/java/org/cirdles/topsoil/app/CloseTabEventHandler.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015 CIRDLES.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.cirdles.topsoil.app;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import javafx.event.Event;
+import javafx.event.EventHandler;
+import javafx.scene.control.Alert;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.Tab;
+import org.cirdles.topsoil.dataset.DatasetResource;
+import org.cirdles.topsoil.dataset.DatasetResourceManager;
+
+/**
+ *
+ * @author parizotclement
+ */
+public class CloseTabEventHandler implements EventHandler<Event> {
+
+    private final Tab tab;
+    private final DatasetResourceManager datasetResourceManager;
+    private final Consumer<Boolean> saveDataTable;
+
+    public CloseTabEventHandler(Tab tab, DatasetResourceManager datasetResourceManager, Consumer<Boolean> callback) {
```

Comment:
Well yes they are. I need the access the Tab to handle the closing event of said Tab. 
I also need to check if the tab is empty or not, and for that I need the DatasetResourceManager. I could do it in the TopsoilMainWindow but you've asked me not to put too much into it.
Same with the callback method.